### PR TITLE
gh-3505 Make sending of OAuth Token for each CTR Launch more configurable

### PR DESCRIPTION
**Pull request by ghillert**
resolves gh-3505

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java [hunk 1]
```diff
@@ -196,23 +196,28 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 
 		if (taskExecutionInformation.isComposed()) {
 			boolean containsAccessToken = false;
+			boolean useUserAccessToken = false;
 
-			final String dataflowAccessTokenKey = "dataflow-server-access-token";
+			final String dataflowServerAccessTokenKey = "dataflow-server-access-token";
+			final String dataflowServerUseUserAccessToken = "dataflow-server-use-user-access-token";
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java]: This new key should somewhat get documented somewhere in `spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc`.

Comment (ghillert) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java]: Added docs.

**Comment by ghillert** [comment]
Added docs + rebased.

**Comment by jvalkeal** [comment]
lgtm, may need to wait for master to become green before merging.

**Pull request closed without merging**